src/Components/ServiceScene/LogOptionsScene.tsx

```diff
@@ -1,30 +1,47 @@
 import { css } from '@emotion/css';
-import { SceneComponentProps, SceneObjectBase, SceneObjectState, sceneGraph } from '@grafana/scenes';
-import { Button, InlineField, InlineSwitch, Tooltip } from '@grafana/ui';
-import React, { ChangeEvent } from 'react';
-import { getLogOption, setLogOption } from 'services/store';
+import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
+import { Button, InlineField, RadioButtonGroup, Tooltip, useStyles2 } from '@grafana/ui';
+import React from 'react';
+import { getLogOption, LogsVisualizationType, setLogOption } from 'services/store';
 import { LogsListScene } from './LogsListScene';
 import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from 'services/analytics';
+import { LogsPanelHeaderActions } from '../Table/LogsHeaderActions';
+import { GrafanaTheme2, LogsSortOrder } from '@grafana/data';
 
 interface LogOptionsState extends SceneObjectState {
-  wrapLines?: boolean;
+  wrapLogMessage?: boolean;
+  visualizationType: LogsVisualizationType;
+  onChangeVisualizationType: (type: LogsVisualizationType) => void;
+  sortOrder?: LogsSortOrder;
 }
 
+/**
+ * The options rendered in the logs panel header
+ */
 export class LogOptionsScene extends SceneObjectBase<LogOptionsState> {
   static Component = LogOptionsRenderer;
 
-  constructor(state?: Partial<LogOptionsState>) {
+  constructor(state: LogOptionsState) {
+    const logsSortOrderFromLocalStorage = getLogOption('sortOrder');
```

Comment:
We could change `getLogOption` to `getLogOption<T>(localStorageKey: string, defaultValue: T)` to return `defaultValue` if it's not found. WDYT?

Reply:
Good call, added